patchwork-rendering/src/main/java/net/minecraftforge/client/event/GuiScreenEvent.java

```diff
@@ -0,0 +1,786 @@
+ * Copyright (c) 2016-2019, 2019
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.client.event;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Consumer;
+
+import net.minecraftforge.eventbus.api.Event;
+import org.lwjgl.glfw.GLFW;
+
+import net.minecraft.client.gui.Element;
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.gui.screen.ingame.AbstractInventoryScreen;
+import net.minecraft.client.gui.widget.AbstractButtonWidget;
+import net.minecraft.client.gui.widget.ButtonWidget;
+
+import net.fabricmc.api.EnvType;
+import net.fabricmc.api.Environment;
+
+/**
+ * Event classes for GuiScreen events.
+ *
+ * @author bspkrs
+ */
+@Environment(EnvType.CLIENT)
+public class GuiScreenEvent extends Event {
+	private final Screen gui;
+
+	// For EventBus
+	public GuiScreenEvent() {
+		this(null);
+	}
+
+	public GuiScreenEvent(Screen gui) {
+		this.gui = gui;
+	}
+
+	/**
+	 * The Screen object generating this event.
+	 */
+	public Screen getGui() {
+		return gui;
+	}
+
+	public static class InitGuiEvent extends GuiScreenEvent {
+		private Consumer<AbstractButtonWidget> add;
+		private Consumer<AbstractButtonWidget> remove;
+
+		private List<AbstractButtonWidget> list;
+
+		// For EventBus
+		public InitGuiEvent() {
+		}
+
+		public InitGuiEvent(Screen gui, List<AbstractButtonWidget> list, Consumer<AbstractButtonWidget> add, Consumer<AbstractButtonWidget> remove) {
+			super(gui);
+			this.list = Collections.unmodifiableList(list);
+			this.add = add;
+			this.remove = remove;
+		}
+
+		/**
+		 * Unmodifiable reference to the list of buttons on the {@link #gui}.
+		 */
+		public List<AbstractButtonWidget> getWidgetList() {
+			return list;
+		}
+
+		public void addWidget(AbstractButtonWidget button) {
+			add.accept(button);
+		}
+
+		public void removeWidget(AbstractButtonWidget button) {
+			remove.accept(button);
+		}
+
+		/**
+		 * <p>This event fires just after initializing {@link Screen#minecraft}, {@link Screen#font},
+		 * {@link Screen#width}, and {@link Screen#height}.</p>
+		 *
+		 * <p>If canceled the following lines are skipped in {@link Screen#init(net.minecraft.client.MinecraftClient, int, int)}:</p>
+		 * {@code this.buttons.clear();}<br/>
```

Comment:
should `<br/>` be used here?

Reply:
Sometimes I wonder if we really have to clean up Forge's JavaDoc mess since this is just for a compatibility layer and no one would actually read these. But I have fixed the JavaDoc issues now.